File: python/src/Creational/Prototype/SaladSpoon.py

#//SaladSpoon.java - One of Two Concrete Prototypes extending the AbstractSpoon Prototype
define ['Creational/Prototype/AbstractSpoon'
],(
AbstractSpoon
) ->

	class SaladSpoon extends AbstractSpoon

		constructor : ->
			@setSpoonName 'Salad Spoon'

	return SaladSpoon

File: go/creational/prototype/SaladSpoon.go

package prototype


//SaladSpoon - One of Two Concrete Prototypes extending the AbstractSpoon Prototype

type SaladSpoon struct {   // implements Utensil
	Namer
}

func NewSaladSpoon() *SaladSpoon {
	return &SaladSpoon{ Namer{ "Salad Spoon" } }
}
